File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-10.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1

        while l <= r:
            mid = l + (r-l) // 2
            if nums[mid-1] > nums[mid] and nums[(mid+1) %len(nums)] > nums[mid]:
                return nums[mid]

            if nums[mid] > nums[r]:
                l = mid + 1
            else:
                r = mid - 1
        return nums[l]

File: Data Structures & Algorithms/two-integer-sum/submission-10.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map();
        
        for(let x = 0; x < nums.length; x++) {
            const comp = target - nums[x];

            if(map.has(comp)) {
                return [map.get(comp), x];
            }

            map.set(nums[x], x);
        }

        return [];
    }
}
